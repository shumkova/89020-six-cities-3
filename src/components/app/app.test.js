import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import App from "./app";

const mockStore = configureStore([]);

const OFFERS = [{
  img: `img/apartment-01.jpg`,
  title: `Beautiful & luxurious apartment at great location`,
  price: 120,
  type: `Appartment`,
  point: [52.3909553943508, 4.85309666406198],
  city: `Amsterdam`,
}];

it(`Render App`, () => {
  const store = mockStore({
    city: `Amsterdam`,
    offers: OFFERS,
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            onCityClick={() => {}}
          />
        </Provider>, {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
