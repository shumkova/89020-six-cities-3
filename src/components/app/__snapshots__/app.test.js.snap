// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render "pending" 1`] = `"pending"`;

exports[`App should render everything 1`] = `"pending"`;
